### PR TITLE
Bed Exciting Palpable Improvement Service ( B.E.P.I.S )

### PR DESCRIPTION
**Pull request by alsoandanswer**
Improves Bed Sprite.

Adds functionality to beds.

### CODE REVIEW

File: code/game/objects/structures/bedsheet_bin.dm [hunk 1]
```diff
@@ -10,22 +10,132 @@ LINEN BINS
...
+	if(inuse)
+		user.show_message("Someone's already using \the [src].")
+		return FALSE
+	inuse = TRUE
+	if (do_after(user, 6, src))
+		if(user.loc != loc)
+			user.do_attack_animation(src)
+		playsound(get_turf(loc), "rustle", 15, 1, -5)
+		if(!message)
+			user.visible_message("<span class='notice'>\The [user] [rolled ? "unrolled" : "rolled"] \the [src].", "You [rolled ? "unrolled" : "rolled"] \the [src].</span>")
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/bedsheet_bin.dm]:
```suggestion
			user.visible_message("<span class='notice'>\The [user] [rolls ? "unrolls" : "rolls"] \the [src].", "You [roll ? "unroll" : "roll"] \the [src].</span>")
```

**Comment by alsoandanswer** [comment]
I ported this code from Eris so there's a buncha fat I gotta trim off.
Thanks for the tips, @NonQueueingMatt

File: code/game/objects/structures/bedsheet_bin.dm [hunk 2]
```diff
@@ -10,22 +10,132 @@ LINEN BINS
...
-		layer = 5
-	else
-		layer = initial(layer)
+/obj/item/weapon/bedsheet/attack_hand(mob/user as mob)
+	if(!folded)
+		toggle_roll(user)
 	add_fingerprint(user)
+
+/obj/item/weapon/bedsheet/MouseDrop(mob/user as mob)
+	if((user == usr && (!( usr.restrained() ) && (!( usr.stat ) && (usr.contents.Find(src) || in_range(src, usr))))))
```
Comment (MarinaGryphon) [inline code/game/objects/structures/bedsheet_bin.dm]:
```suggestion
	if((user && (!( user.restrained() ) && (!( user.stat ) && (src in user.contents || in_range(src, user))))))
```

File: code/game/objects/structures/bedsheet_bin.dm [hunk 3]
```diff
@@ -10,22 +10,132 @@ LINEN BINS
...
+		if(!istype(usr, /mob/living/carbon/slime) && !istype(usr, /mob/living/simple_animal))
```
Comment (MarinaGryphon) [inline code/game/objects/structures/bedsheet_bin.dm]:
```suggestion
		if(!istype(user, /mob/living/carbon/slime) && !istype(user, /mob/living/simple_animal))
```

File: code/game/objects/structures/bedsheet_bin.dm [hunk 4]
```diff
@@ -10,22 +10,132 @@ LINEN BINS
...
+			if( !usr.get_active_hand() )		//if active hand is empty
```
Comment (MarinaGryphon) [inline code/game/objects/structures/bedsheet_bin.dm]:
```suggestion
			if( !user.get_active_hand() )		//if active hand is empty
```

File: code/game/objects/structures/bedsheet_bin.dm [hunk 5]
```diff
@@ -10,22 +10,132 @@ LINEN BINS
...
+				var/mob/living/carbon/human/H = user
+				var/obj/item/organ/external/temp = H.organs_by_name["r_hand"]
+				if (H.hand)
+					temp = H.organs_by_name["l_hand"]
+				if(temp && !temp.is_usable())
+					to_chat(user, "<span class='notice'>You try to move your [temp.name], but cannot!</span>")
+					return
+				if(!folded)
+					toggle_fold(usr)
```
Comment (MarinaGryphon) [inline code/game/objects/structures/bedsheet_bin.dm]:
```suggestion
					toggle_fold(user)
```

File: code/game/objects/structures/bedsheet_bin.dm [hunk 6]
```diff
@@ -10,22 +10,132 @@ LINEN BINS
...
+/obj/item/weapon/bedsheet/verb/roll_verb(mob/living/user as mob)
+	set name = "Roll bedsheet"
+	set category = "Object"
+	set src in view(1)
+
+	if(folded)
+		user.show_message("<span class='warning'>Unfold \the [src] first.</span>")
+	else if(istype(loc,/mob))
+		user.show_message("<span class='warning'>Drop \the [src] first.</span>")
+	else if(ishuman(usr))
```
Comment (MarinaGryphon) [inline code/game/objects/structures/bedsheet_bin.dm]:
```suggestion
	else if(ishuman(user))
```

File: code/game/objects/structures/bedsheet_bin.dm [hunk 7]
```diff
@@ -10,22 +10,132 @@ LINEN BINS
...
+		toggle_roll(usr)
```
Comment (MarinaGryphon) [inline code/game/objects/structures/bedsheet_bin.dm]:
```suggestion
		toggle_roll(user)
```

File: code/game/objects/structures/bedsheet_bin.dm [hunk 8]
```diff
@@ -10,22 +10,132 @@ LINEN BINS
...
+	return FALSE
+
+/obj/item/weapon/bedsheet/verb/fold_verb(mob/living/user as mob)
+	set name = "Fold bedsheet"
+	set category = "Object"
+	set src in view(1)
+
+	if(istype(loc,/mob))
+		user.show_message("<span class='warning'>Drop \the [src] first.</span>")
+	else if(ishuman(usr))
```
Comment (MarinaGryphon) [inline code/game/objects/structures/bedsheet_bin.dm]:
```suggestion
	else if(ishuman(user))
```

File: code/game/objects/structures/bedsheet_bin.dm [hunk 9]
```diff
@@ -10,22 +10,132 @@ LINEN BINS
...
+		toggle_fold(usr)
```
Comment (MarinaGryphon) [inline code/game/objects/structures/bedsheet_bin.dm]:
```suggestion
		toggle_fold(user)
```

File: code/game/objects/structures/stool_bed_chair_nest/bed.dm [hunk 10]
```diff
@@ -173,7 +173,7 @@
 	..(newloc,"wood","leather")
 
 /obj/structure/bed/padded/New(var/newloc)
-	..(newloc,"plastic","cotton")
+	..(newloc,DEFAULT_WALL_MATERIAL,"cotton")
```
Comment (MarinaGryphon) [inline code/game/objects/structures/stool_bed_chair_nest/bed.dm]: Why change it from plastic to steel?

Comment (alsoandanswer) [reply code/game/objects/structures/stool_bed_chair_nest/bed.dm]: Makes them in line with the rest of the furniture.

**Pull request closed without merging**